### ISSUE TITLE
Domain Picker: add jest-axe for a11y testing

### ISSUE THREAD

**Issue by ciampo** [Automattic/wp-calypso#51142]
## What

Use axe (an a11y testing tool) in the Domain Picker automated tests by using the [jest-axe](https://github.com/nickcolley/jest-axe) integration

## Why

This will help spotting the most obvious a11y issues in our code just by adding one additional insertion in any tests.

## A/C

- [ ] Add `jest-axe` to Calypso's `dev` dependencies
- [ ] Add `jest-axe` to the Domain Picker package's jest configuration
- [ ] Add Axe configuration options (e.g. what type of a11y issues should be flagged)
- [ ] Add axe expectation/assertions in current tests
    - [ ] As axe assertion fails, we should open separate issue for every failure

## Context

Later on, when adding Storybook to the package, we should consider adding the `a11y` plugin and sharing the same Axe configuration with the `jest-axe` plugin

**Comment by youbrokesomething** [comment]
The jest-axe package and the axe-playwright packages are going to be added as part of my improving accessibility on Dotcom projcet.
